src/configure/clients/devOps/azureDevOpsClient.ts

```diff
@@ -135,8 +207,77 @@ export class AzureDevOpsClient {
         }
     }
 
+    public async validateOrganizationName(organizationName: string): Promise<string> {
+        let deferred = Q.defer<string>();
+        let accountNameRegex = new RegExp(/^[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]$|^[a-zA-Z]$/);
+
+        if(!organizationName || /^\\s/.test(organizationName) || /\\s$/.test(organizationName) || organizationName.indexOf("-") === 0 || !accountNameRegex.test(organizationName)) {
+            deferred.resolve(Messages.organizationNameStaticValidationMessage);
+        }
+
+        if(ReservedHostNames.indexOf(organizationName) >= 0) {
+            deferred.resolve(util.format(Messages.organizationNameReservedMessage, organizationName));
+        }
+        
+        let url = `https://app.vsaex.visualstudio.com/_apis/HostAcquisition/NameAvailability/${organizationName}`;
+
+        this.restClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: url,
+            headers: {
+                "Content-Type": "application/json",
+                "Accept": "api-version=5.0-preview.1"
+            },
+            method: "GET",
+            deserializationMapper: null,
+            serializationMapper: null
+        })
+        .then((response) => {
+            if( response.name === organizationName && !response.isAvailable) {
+                deferred.resolve(util.format(Messages.organizationNameReservedMessage, organizationName));
+            }
+            deferred.resolve("");
+        })
+        .catch(() => {
+            deferred.resolve("");
+        });
+
+        return deferred.promise;
+    }
+
+    public async getProjectIdFromName(organizationName: string, projectName: string): Promise<string> {
```

Comment:
Yes. As discussed offline, will try to do it in a neat way if it doesn't make the logic complex/messy.

Reply:
In case of new project, the POST call returns operation status. We need to do a GET call on project after operation monitoring is complete. So we need to make this call. However, in case of existing project we can store the id along with name while getting project list,